client/app/components/notification/notification.controller.js

```diff
@@ -0,0 +1,36 @@
+class NotificationController {
+  constructor(User, Authentication, $state, $injector) {
+    "ngInject";
+    Object.assign(this, {
+      User,
+      Authentication,
+      $state,
+      $injector
+    });
+  }
+
+  isVerified() {
+    return this.userdata.mail_verified;
+  }
+  sendVerification(email) {
+    const $mdToast = this.$injector.get("$mdToast");
+    this.Authentication.update().then((data) => {
+      if (data.mail_verified === false) {
+        this.Authentication.verify(email).then(() => {
+          $mdToast.show($mdToast.simple()
+            .textContent("Verification Email Sent")
+            .highlightAction(false));
+          return false;
+        }).catch(() => {
+          $mdToast.show($mdToast.simple()
+            .textContent("Email does not Exist")
```

Comment:
There are three cases where this could happen:

- connection interrupted
- server error
- mail already verified

The first two are already handled in a $http interceptor (see `app.HTTPErrorHandler.js` and the last one is not expected.

I would either leave this away or just forward the error message of the server:

```
catch((err) => {
...
.textContent(err)
...
}
```

It will probably be a ugly message, but it shouldn't happen at all ;)